python/cudf_polars/cudf_polars/experimental/parallel.py

```diff
@@ -261,9 +261,14 @@ def _(
     ir: Union, rec: LowerIRTransformer
 ) -> tuple[IR, MutableMapping[IR, PartitionInfo]]:
     # Check zlice
-    if ir.zlice is not None:  # pragma: no cover
-        return _lower_ir_fallback(
-            ir, rec, msg="zlice is not supported for multiple partitions."
+    if ir.zlice is not None:
+        offset, length = ir.zlice
+        if length is None:  # pragma: no cover
+            return _lower_ir_fallback(
+                ir, rec, msg="This slice is not supported for multiple partitions."
+            )
```

Comment:
```suggestion
```
Sorry! I got this completely wrong before. The problem with `Union` is that `_lower_ir_fallback` doesn't work when you need to reduce the number of children (only when you want multi-partition children to be repartitioned)!

If we remove the fallback case, we will allow the `Slice` `lower_ir_node` implementation call `_lower_ir_fallback` (which should work fine).

Reply:
Done https://github.com/rapidsai/cudf/pull/19336/commits/536f7aa77235c721b08e0ccbed7b59483449ab18